chalice/awsclient.py

```diff
@@ -154,8 +155,8 @@ def update_function(self,
             kwargs['Role'] = role_arn
         if kwargs:
             kwargs['FunctionName'] = function_name
-            self._create_or_update_function_with_retries(
-                'update_function_configuration', kwargs)
+            self._call_client_method_with_retries(
+                self._client('lambda').update_function_configuration, kwargs)
```

Comment:
You already create a lambda_client in the first line of this function, you shouldn't need to create it again here.